src/rpm/headers/types.rs

```diff
@@ -39,17 +39,19 @@ pub enum FileMode {
     // is the bit representation which is the same for both.
     Dir { permissions: u16 },
     Regular { permissions: u16 },
+    SymbolicLink { permissions: u16 },
     // For "Invalid" we use a larger integer since it is possible to create an invalid
     // FileMode by providing an overflowing integer.
     Invalid { raw_mode: i32, reason: &'static str },
 }
 
-// there are more file types but in the context of RPM, only regular and directory should be relevant.
+// there are more file types but in the context of RPM, only regular and directory and symbolic file should be relevant.
 // See <https://man7.org/linux/man-pages/man7/inode.7.html> section "The file type and mode"
 const FILE_TYPE_BIT_MASK: u16 = 0o170000; // bit representation = "1111000000000000"
 const PERMISSIONS_BIT_MASK: u16 = 0o7777; // bit representation = "0000111111111111"
-const REGULAR_FILE_TYPE: u16 = 0o100000; //  bit representation = "1000000000000000"
-const DIR_FILE_TYPE: u16 = 0o040000; //      bit representation = "0100000000000000"
+pub const REGULAR_FILE_TYPE: u16 = 0o100000; //  bit representation = "1000000000000000"
+pub const DIR_FILE_TYPE: u16 = 0o040000; //      bit representation = "0100000000000000"
+pub const SYMBOLIC_LINK_FILE_TYPE: u16 = 0o120000; // bit representation = "1010000000000000"
```

Comment:
I kind of feel like this needs a pass to figure out how to improve these APIs such that exposing it isn't necessary.  But it doesn't need to be part of your PR, I can look into it later (unless you want to)

Reply:
Yes, it seems we have some works to do, I will continuous contribute to this project if I have some ideas.